node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -170,31 +204,44 @@ func (atp *apiTransactionProcessor) fetchTxsForSender(sender string, senderShard
 		txsForSender = append(txsForSender, txs...)
 	}
 
+	sort.Slice(txsForSender, func(i, j int) bool {
+		return txsForSender[i].Tx.GetNonce() < txsForSender[j].Tx.GetNonce()
+	})
+
 	return txsForSender
 }
 
-func (atp *apiTransactionProcessor) fetchLastNonceForSender(sender string, senderShard uint32) (uint64, bool) {
-	lastNonce := uint64(0)
-	found := false
-	cachers := atp.getDataStoresForSender(senderShard)
-	for _, cache := range cachers {
-		txCache, ok := cache.(*txcache.TxCache)
-		if !ok {
-			continue
-		}
+func (atp *apiTransactionProcessor) fetchLastNonceForSender(sender string, senderShard uint32) (uint64, error) {
+	wrappedTxs := atp.fetchTxsForSender(sender, senderShard)
+	if len(wrappedTxs) == 0 {
+		return 0, fmt.Errorf("%w, no transaction in pool for sender", ErrCannotRetrieveTransactions)
+	}
 
-		lastShardNonce, ok := txCache.GetLastPoolNonceForSender(sender)
-		if !ok {
-			continue
-		}
+	lastTx := wrappedTxs[len(wrappedTxs)-1]
+	return lastTx.Tx.GetNonce(), nil
+}
+
+func (atp *apiTransactionProcessor) extractNonceGaps(sender string, senderShard uint32) ([]common.NonceGapApiResponse, error) {
+	wrappedTxs := atp.fetchTxsForSender(sender, senderShard)
+	if len(wrappedTxs) == 0 {
+		return nil, fmt.Errorf("%w, no transaction in pool for sender", ErrCannotRetrieveTransactions)
+	}
 
-		if lastShardNonce > lastNonce {
-			lastNonce = lastShardNonce
-			found = true
+	nonceGaps := make([]common.NonceGapApiResponse, 0)
+	for i := 0; i < len(wrappedTxs)-1; i++ {
+		nextNonce := wrappedTxs[i+1].Tx.GetNonce()
+		currentNonce := wrappedTxs[i].Tx.GetNonce()
+		nonceDiff := nextNonce - currentNonce
+		if nonceDiff > 1 {
+			nonceGap := common.NonceGapApiResponse{
+				From: fmt.Sprintf("%d", currentNonce),
+				To:   fmt.Sprintf("%d", nextNonce),
+			}
```

Comment:
wouldn't it be better to use integer type for `From` and `To` since they represent nonce values?

Reply:
indeed..changed